crates/sui-framework/packages/sui-framework/sources/crypto/groth16.move

```diff
@@ -60,8 +65,11 @@ module sui::groth16 {
         bytes: vector<u8>,
     }
 
-    /// Creates a `PublicProofInputs` wrapper from bytes.
+    /// Creates a `PublicProofInputs` wrapper from bytes. The `bytes` parameter should be a concatenation of a number of
+    /// 32 bytes scalar field elements to be used as public inputs in little-endian format to a circuit.
     public fun public_proof_inputs_from_bytes(bytes: vector<u8>): PublicProofInputs {
+        assert!(bytes.length() % 32 == 0, EInvalidScalar);
```

Comment:
what happens in the current code when this check is false?

Reply:
An "invalid input" error is returned from fastcrypto.